### PR TITLE
remove duplicate json schema tests

### PR DESCRIPTION
**Pull request by graphaelli**
removes `Test*ProcessorValidationFailed` duplicate of a test in the respective `*/package_tests/json_schema_test.go`

### CODE REVIEW

File: tests/json_schema.go [hunk 1]
```diff
@@ -158,6 +158,8 @@ func TestDataAgainstProcessor(t *testing.T, p processor.Processor, testData []Sc
 		data, err := loader.LoadData(d.File)
 		assert.Nil(t, err)
 		err = p.Validate(data)
-		assert.Contains(t, err.Error(), d.Error)
+		if assert.Error(t, err) {
```
Comment (jalvz) [inline tests/json_schema.go]: how this is removing duplicates?

**Comment by jalvz** [comment]
I don't see how this removes duplicated tests. The jsons in `valid` are positive tests (must validate), and the ones in `invalid` are negative tests (must not validate).
Am I missing something?

File: tests/json_schema.go (hunk 1 above)
Comment (simitt) [inline tests/json_schema.go]: I generally agree that the tests are duplicates. However, the removed tests did check that `p.Validate(data)` returns an error. Removing those tests and changing this check to `if assert.Error(t, err).. ` we lose the check that the validation fails.

Comment (graphaelli) [reply tests/json_schema.go]: That is confusing.  It doesn't actually lost the check that that validation fails, it just prevents the `assert.Contains` from panicking when err is actually nil.

Comment (graphaelli) [reply tests/json_schema.go]: ```An error is expected but got nil.``` is produced in that case

Comment (simitt) [reply tests/json_schema.go]: ah sry didn't get that.

**Pull request merged**